weave/trace/weave_client.py

```diff
@@ -143,13 +143,26 @@ class Call:
     output: Any = None
     exception: Optional[str] = None
     summary: Optional[dict] = None
-    display_name: Optional[str] = None
+    display_name: Optional[Union[str, Callable[["Call"], str]]] = None
     attributes: Optional[dict] = None
     # These are the live children during logging
     _children: list["Call"] = dataclasses.field(default_factory=list)
 
     _feedback: Optional[RefFeedbackQuery] = None
 
+    @property
+    def func_name(self) -> str:
+        """
+        The decorated function's name that produced this call.
+
+        This is different from `op_name` which is usually the ref of the op.
+        """
+        if self.op_name.startswith("weave:///"):
+            ref = parse_uri(self.op_name)
```

Comment:
Note: `parse_uri` does throw under certain conditions where the ref string is not parsable. (Shouldn't occur often, but technically is possible)

Reply:
I can live with that here.  If we detect errors here I'll add more handling